CMakeLists.txt

```diff
@@ -29,6 +29,19 @@ if(COMMAND cmake_policy)
     if(POLICY CMP0060)
         cmake_policy(SET CMP0060 NEW) # Full lib paths.
     endif()
+
+    if(POLICY CMP0091)
+        cmake_policy(SET CMP0091 NEW) # Do not set MSVC runtime library flags.
+    endif()
+
+    if(POLICY CMP0092)
+        cmake_policy(SET CMP0092 NEW) # Do not set /w for MSVC.
+    endif()
+
+    if(POLICY CMP0117)
+        cmake_policy(SET CMP0117 NEW) # Do not set /GR for MSVC.
+    endif()
+
```

Comment:
So much nonsense. Just put the `cmake_minimum_required` at the top of the CMakeList and set the version to something reasonable. This monstrosity of build script is already using at least 3.1 features.